src/utils/model.js

```diff
@@ -1,157 +1,91 @@
-      ModelIds.baseline,
-      ModelIds.strictDistancingNow,
-      ModelIds.projected,
-      ModelIds.weakDistancingNow,
-    ];
-  } else {
-    projectionsToLoad = [
-      ModelIds.baseline,
-      ModelIds.strictDistancingNow,
-      ModelIds.strictDistancingNow, // TODO(igor): HAX!
-      ModelIds.weakDistancingNow,
-    ];
-  }
-  let urls = projectionsToLoad.map(i => {
-    let fipsCode =
-      county && county.full_fips_code ? county.full_fips_code : null;
-    const stateUrl = `${dataUrl}/${location}.${i}.json`;
-    const countyUrl = `${dataUrl}/county/${location.toUpperCase()}.${fipsCode}.${i}.json`;
-    return county ? countyUrl : stateUrl;
-  });
-
-  try {
-    let loadedModelDatas = await fetchAll(urls);
-    // HACK: Truncate data to 32 data points to make new models match old models.
-    loadedModelDatas = loadedModelDatas.map(data => data.slice(0, 32));
-    //This is to fix county data format
-    modelDataForKey = {
-      projections: new Projections(loadedModelDatas, location, county),
-    };
-  } catch (err) {
-    modelDataForKey = {
-      error: true,
-      payload: err,
-    };
-  }
-
-  return modelDataForKey;
-}
-
-async function fetchDataAndSet(
-  setModelDatas,
-  location,
-  county = null,
-  dataUrl = null,
+async function fetchProjections(
+  stateAbbr,
+  countyInfo = null,
+  dataUrl = DATA_URL,
 ) {
-  const key = county ? 'countyDatas' : 'stateDatas';
-
-  const modelDataForKey = await fetchData(location, county, dataUrl);
-
-  setModelDatas(m => {
-    return {
-      ...m,
-      location,
-      county,
-      [key]: modelDataForKey,
-    };
-  });
+  const fileIdToIntervention = {
+    0: INTERVENTIONS.LIMITED_ACTION,
+    1: INTERVENTIONS.SHELTER_IN_PLACE,
+    2: INTERVENTIONS.PROJECTED,
+    3: INTERVENTIONS.SOCIAL_DISTANCING,
+  };
+  const fileIdsToLoad = Object.keys(fileIdToIntervention);
+
+  // load all the projections for a state or county in parallel
+  const countyProjectionUrl = i =>
+    `${dataUrl}/county/${stateAbbr.toUpperCase()}.${
+      countyInfo.full_fips_code
+    }.${i}.json`;
+  const stateProjectionUrl = i => `${dataUrl}/${stateAbbr}.${i}.json`;
+  const urls = countyInfo
+    ? fileIdsToLoad.map(i => countyProjectionUrl(i))
+    : fileIdsToLoad.map(i => stateProjectionUrl(i));
+  let projectionsData = await fetchAll(urls);
+
+  // HACK: Truncate data to 32 data points to make new models match old models.
+  projectionsData = projectionsData.map(data =>
+    data ? data.slice(0, 32) : null,
+  );
+
+  // annotate each dataset with the intervention
+  const projectionInfos = projectionsData.map((pd, idx) => ({
+    data: pd,
+    intervention: fileIdToIntervention[fileIdsToLoad[idx]],
+  }));
+
+  return new Projections(projectionInfos, stateAbbr, countyInfo);
 }
 
-export function useModelDatas(location, county = null, dataUrl = null) {
-  const [modelDatas, setModelDatas] = useState(initialData);
+export function useProjections(location, county = null) {
```

Comment:
nit: location => stateId ?

Reply:
let me touch it up in the second refactor (where i convert all uses of location to stateId)